graylog2-web-interface/src/views/components/widgets/MessageList.jsx

```diff
@@ -51,141 +63,94 @@ class MessageList extends React.Component<Props, State> {
+      total: PropTypes.number.isRequired,
+      id: PropTypes.string.isRequired,
+    }).isRequired,
+    effectiveTimerange: PropTypes.shape({
+      from: PropTypes.string.isRequired,
+      to: PropTypes.string.isRequired,
+      type: PropTypes.string.isRequired,
     }).isRequired,
-    containerHeight: PropTypes.number,
     selectedFields: PropTypes.object,
     currentView: PropTypes.object,
   };
 
   static defaultProps = {
     pageSize: Messages.DEFAULT_LIMIT,
-    containerHeight: undefined,
     selectedFields: Immutable.Set(),
     currentView: { view: {}, activeQuery: undefined },
     config: undefined,
   };
 
-  static contextType = RenderCompletionCallback;
-
   state = {
+    errors: [],
     currentPage: 1,
-    expandedMessages: Immutable.Set(),
   };
 
   componentDidMount() {
     const onRenderComplete = this.context;
     InputsActions.list().then(() => (onRenderComplete && onRenderComplete()));
   }
 
-  _getSelectedFields = () => {
-    const { selectedFields, config } = this.props;
-    if (config) {
-      return Immutable.Set(config.fields);
-    }
-    return selectedFields;
-  };
 
-  _columnStyle = (fieldName) => {
-    const { fields } = this.props;
-    const selectedFields = Immutable.OrderedSet(fields);
-    if (fieldName.toLowerCase() === 'source' && selectedFields.size > 1) {
-      return { width: 180 };
+  _resultWindowLimitMessage = (errors = []) => {
+    const { pageSize } = this.props;
+    const resultWindowLimitError = errors.find(error => error.resultWindowLimit);
+    if (resultWindowLimitError) {
+      const { resultWindowLimit } = resultWindowLimitError;
+      const validPages = Math.floor(resultWindowLimit / pageSize);
+      return { description: `Elasticsearch limits the search result to ${resultWindowLimit} messages. With a page size of ${pageSize} messages, you can use the first ${validPages} pages.` };
     }
-    return {};
-  };
+    return undefined;
+  }
 
-  _toggleMessageDetail = (id) => {
-    let newSet;
-    const { expandedMessages } = this.state;
-    if (expandedMessages.contains(id)) {
-      newSet = expandedMessages.delete(id);
-    } else {
-      newSet = expandedMessages.add(id);
-      RefreshActions.disable();
-    }
-    this.setState({ expandedMessages: newSet });
-  };
+  _handlePageChange = (pageNo: number) => {
+    // execute search with new offset
+    const { pageSize, data: { id: searchTypeId }, effectiveTimerange } = this.props;
+    const searchTypePayload = { [searchTypeId]: { limit: pageSize, offset: pageSize * (pageNo - 1) } };
+    RefreshActions.disable();
+    SearchActions.reexecuteSearchTypes(searchTypePayload, effectiveTimerange).then((response) => {
+      let errors = [...response.result.errors];
+      if (!isEmpty(errors)) {
+        const validPagesInfo = this._resultWindowLimitMessage(response.result.errors);
+        if (validPagesInfo) {
+          errors = [
+            validPagesInfo,
+            ...errors,
+          ];
+        }
```

Comment:
Is it okay to just add a custom error, which has only a description? We are only doing this to inform the user about the working / valid pages.